### PR TITLE
Tracking 2.0.0

### PR DESCRIPTION
**Pull request by mstriemer**
@ngokevin I'm going to merge this along with https://github.com/mozilla/fireplace/pull/1036 and https://github.com/mozilla/transonic/pull/92 shortly. Any other comments?

### CODE REVIEW

File: core/router.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+define('core/router', [
```
Comment (ngokevin) [inline core/router.js]: this file could use some comments (docstring + what's happening in addRoute). requires don't need to be on different lines.

File: core/router.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                _.extend(exports.api.routes, routes);
+            },
+            clearRoutes: function() {
+                exports.api.routes = {};
+            },
+            routes: {},
+        },
+    };
+
+    return exports;
```
Comment (ngokevin) [inline core/router.js]: not just return above?

File: core/settings.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+define('core/settings', ['underscore'], function(_) {
+    var settings = {
```
Comment (ngokevin) [inline core/settings.js]: can we return here

File: core/builder.js [hunk 4]
```diff
@@ -3,10 +3,10 @@
...
     rendering, in-place pagination.
 */
-define('builder',
-    ['log', 'jquery', 'templates', 'models', 'requests', 'settings', 'z',
-     'nunjucks.compat'],
-    function(log, $, nunjucks, models, requests, settings, z) {
+define('core/builder',
+    ['core/log', 'core/nunjucks', 'core/models', 'core/requests', 'core/settings', 'core/z',
+     'core/nunjucks.compat', 'jquery'],
+    function(log, nunjucks, models, requests, settings, z, nunjucksCompat, $) {
```
Comment (ngokevin) [inline core/builder.js]: i like adding line breaks to the dependency injection line/function arguments to match the line breaks in the dependency list. Makes it easier to match up when removing/adding modules.

```
['core/log', ... 'core/z',
 'core/nunjucks.compat', 'jquery']
function(log, ... z, 
         nunjucksCompat, $)
```

**Comment by ngokevin** [comment]
have some nits about how we format our AMD defines (module name, then deps on a separate line with line breaks at 80 chars, then the callback on a different line with arguments matching the line breaks of the second line). not sure whether i should let it go

don't forget to do marketplace-template and marketplace-frontend-docs

r+wc, trust this works :)

File: core/router.js (hunk 2 above)
Comment (mstriemer) [inline core/router.js]: I did this so that I can refer to the object without using `this`.

File: core/settings.js (hunk 3 above)
Comment (mstriemer) [inline core/settings.js]: This is also to allow referring to the object without using `this`.

**Pull request merged**